lib/AccessTokenClient.ts

```diff
@@ -80,18 +94,24 @@ export class AccessTokenClient {
     request.user_pin = pin;
 
     if (issuanceInitiationRequest[PRE_AUTH_CODE_LITERAL]) {
+      if (codeVerifier) {
+        throw new Error('Cannot pass a code_verifier when flow type is pre-authorized');
+      }
       request.grant_type = GrantTypes.PRE_AUTHORIZED_CODE;
       request[PRE_AUTH_CODE_LITERAL] = issuanceInitiationRequest[PRE_AUTH_CODE_LITERAL];
     }
     if (issuanceInitiationRequest.op_state) {
       this.throwNotSupportedFlow();
```

Comment:
Okay guess that part still isn't supported. I think the next comment about moving the check for pre-auth code and auth code is even more relevant to not confuse another developer, seeing code after something which clearly throws an error

Reply:
Yes I think I've made a mistake because it's unclear what the `op_state` variable is used for. I assumed the presence of `issuanceInitiationRequest[PRE_AUTH_CODE_LITERAL]` would determine the flow type. Am I wrong here?

Reply:
Yeah, but I guess we will want to refactor that in the future. As the wallet probably will not know upfront which flow it is going to deal with, so we will have to deduct it from the payloads